trulens_eval/trulens_eval/guardrails/base.py

```diff
@@ -0,0 +1,56 @@
+from concurrent.futures import wait
+from trulens_eval.feedback import Feedback
+from trulens_eval.utils.containers import first
+from trulens_eval.utils.containers import second
+from trulens_eval.utils.threading import ThreadPoolExecutor
+from functools import wraps
+
+class context_filter:
+    def __init__(self, feedback: Feedback, threshold: float):
+        self.feedback = feedback
+        self.threshold = threshold
+    """
+    ContextFilter is a class that filters contexts based on a given feedback and threshold.
+
+    Parameters:
+    feedback (Feedback): The feedback object to use for filtering.
+    threshold (float): The minimum feedback value required for a context to be included.
+
+    !!! example
+
+        ```python
+        feedback = Feedback(provider.groundedness_measure_with_cot_reasons, name="Groundedness")
+        @context_filter(feedback, 0.5)
+        def retrieve(query: str) -> list:
+            results = vector_store.query(
+            query_texts=query,
+            n_results=3
+        )
+        return [doc for sublist in results['documents'] for doc in sublist]
+        ```
+    """
+
+    def __call__(self, func):
```

Comment:
Add `@functools.wraps(func)` here

Reply:
Something with `@functools.wraps(func)` seems to break the effect of the wrapper here and stop the filtering, while using the below should have the same effect and retains the effect. open to suggestions here
```
        wrapper.__name__ = func.__name__
        wrapper.__doc__ = func.__doc__
        wrapper.__dict__= func.__dict__
```

Reply:
stop the filtering meaning the unfiltered function gets used and returns all results, not just the filtered ones

Reply:
Can you add a note about this problem so future readers know why we cannot use the expected functools.wraps.